src/classic/element/ReactElement.js

```diff
@@ -228,6 +228,60 @@ ReactElement.cloneAndReplaceProps = function(oldElement, newProps) {
   return newElement;
 };
 
+ReactElement.cloneElement = function(element, config, children) {
+  var propName;
+
+  // Original props are copied
+  var props = assign({}, element.props);
+
+  // Reserved names are extracted
+  var key = element.key;
+  var ref = element.ref;
+
+  // Owner will be preserved, unless ref is overridden
+  var owner = element._owner;
+
+  if (config != null) {
+    if (config.ref !== undefined) {
```

Comment:
No, I thought about that, but since this is very explicitly an override I don't think it is necessary. Just like any other prop overrides silently. We used to have a warning because it was __implicitly__ deleting the ref even if you didn't provide a new one.